### PR TITLE
Allow critical pods to be scheduled on masters [ch23732]

### PR DESCRIPTION
**Pull request by emosbaugh**

### CODE REVIEW

File: install_scripts/templates/kubernetes/yaml/rook-1-0-6-cluster.yml [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+metadata:
+  name: replicapool
+  namespace: rook-ceph
+spec:
+  failureDomain: host
+  replicated:
+    # the REK operator will scale this up to 3 as nodes join the cluster
+    size: 1
+---
+apiVersion: v1
```
Comment (emosbaugh) [inline install_scripts/templates/kubernetes/yaml/rook-1-0-6-cluster.yml]: added this

File: install_scripts/templates/kubernetes/yaml/rook-1-0-6-cluster.yml [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+  namespace: rook-ceph
+---
+apiVersion: ceph.rook.io/v1
+kind: CephCluster
+metadata:
+  name: rook-ceph
+  namespace: rook-ceph
+spec:
+  cephVersion:
+    image: ceph/ceph:v14.2.8-20200305
```
Comment (emosbaugh) [inline install_scripts/templates/kubernetes/yaml/rook-1-0-6-cluster.yml]: new image

File: install_scripts/templates/kubernetes/yaml/rook-1-0-6-system.yml [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: rook-ceph-operator
+    spec:
+      serviceAccountName: rook-ceph-system
+      containers:
+      - name: rook-ceph-operator
+        image: rook/ceph:v1.0.6
```
Comment (emosbaugh) [inline install_scripts/templates/kubernetes/yaml/rook-1-0-6-system.yml]: new image

File: install_scripts/templates/kubernetes/yaml/rook-1-0-3-system.yml [hunk 4]
```diff
@@ -474,11 +474,31 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
+        # Rook Agent toleration. Will tolerate all taints with all keys.
```
Comment (emosbaugh) [inline install_scripts/templates/kubernetes/yaml/rook-1-0-3-system.yml]: i figure these tolerations cant hurt

File: install_scripts/templates/kubernetes/yml-generate.sh [hunk 5]
```diff
@@ -930,8 +958,7 @@ $weave_passwd_env
             seLinuxOptions: {}
           serviceAccountName: weave-net
           tolerations:
-            - effect: NoSchedule
-              operator: Exists
+            - operator: Exists
```
Comment (emosbaugh) [inline install_scripts/templates/kubernetes/yml-generate.sh]: this is a weird one. i maybe should experiment with changing this

File: install_scripts/templates/common/kubernetes.sh [hunk 6]
```diff
@@ -526,7 +526,9 @@ airgapLoadKubernetesCommonImages1153() {
         docker tag 0246380e4b70 docker.io/envoyproxy/envoy-alpine:v1.10.0
         docker tag 672aff19e6e4 gcr.io/heptio-images/contour:v0.13.0
         docker tag 3ee0a397fb56 docker.io/rook/ceph:v1.0.3
+        docker tag TODO docker.io/rook/ceph:v1.0.6
```
Comment (areed) [inline install_scripts/templates/common/kubernetes.sh]: TODO

File: install_scripts/templates/kubernetes/yaml/rook-1-0-3-system.yml [hunk 7]
```diff
@@ -474,11 +474,31 @@ spec:
...
+          value: node-role.kubernetes.io/master
+        # Rook Discover toleration. Will tolerate all taints with all keys.
+        # Choose between NoSchedule, PreferNoSchedule and NoExecute:
+        - name: DISCOVER_TOLERATION
+          value: NoSchedule
+        # (Optional) Rook Discover toleration key. Set this to the key of the taint you want to tolerate
+        - name: DISCOVER_TOLERATION_KEY
+          value: node-role.kubernetes.io/master
+        - name: ROOK_LOG_LEVEL
+          value: DEBUG
```
Comment (areed) [inline install_scripts/templates/kubernetes/yaml/rook-1-0-3-system.yml]: Was it intentional to leave it at debug?

File: install_scripts/templates/kubernetes/yaml/rook-1-0-6-system.yml [hunk 8]
```diff
@@ -0,0 +1,506 @@
...
+          value: node-role.kubernetes.io/master
+        # Rook Discover toleration. Will tolerate all taints with all keys.
+        # Choose between NoSchedule, PreferNoSchedule and NoExecute:
+        - name: DISCOVER_TOLERATION
+          value: NoSchedule
+        # (Optional) Rook Discover toleration key. Set this to the key of the taint you want to tolerate
+        - name: DISCOVER_TOLERATION_KEY
+          value: node-role.kubernetes.io/master
+        - name: ROOK_LOG_LEVEL
+          value: DEBUG
```
Comment (areed) [inline install_scripts/templates/kubernetes/yaml/rook-1-0-6-system.yml]: here too?

File: install_scripts/templates/kubernetes/yml-generate.sh (hunk 5 above)
Comment (areed) [reply install_scripts/templates/kubernetes/yml-generate.sh]: weave pods weren't getting scheduled on masters?

Comment (emosbaugh) [reply install_scripts/templates/kubernetes/yml-generate.sh]: they were not

File: install_scripts/templates/kubernetes/yaml/rook-1-0-3-system.yml (hunk 7 above)
Comment (emosbaugh) [reply install_scripts/templates/kubernetes/yaml/rook-1-0-3-system.yml]: no but maybe i should leave it while in qa on staging

**Pull request merged**